modules/ztp-managing-sriov-vlan-with-hub-cluster-templates-in-pgt.adoc

```diff
@@ -0,0 +1,74 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc
+
+:_content-type: PROCEDURE
+[id="ztp-managing-sriov-vlan-with-hub-cluster-templates-in-pgt_{context}"]
+= Specifying VLAN IDs in group PolicyGenTemplate CRs with hub cluster templates
+
+You can manage VLAN IDs for managed clusters in a single `ConfigMap` CR and use hub cluster templates to populate the VLAN IDs in the generated polices that get applied to the clusters.
+
+The following example shows how you how manage VLAN IDs in single `ConfigMap` CR and apply them in individual cluster polices by using a single `PolicyGenTemplate` group CR.
+
+[NOTE]
+====
+When using the `fromConfigmap` function, the `printf` variable is only available for the template resource `data` key fields.
+You cannot use it with `name` and `namespace` fields.
+====
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+
+* You have logged in to the hub cluster as a user with `cluster-admin` privileges.
+
+* You have created a Git repository where you manage your custom site configuration data.
+The repository must be accessible from the hub cluster and be defined as a source repository for the Argo CD application.
+
+.Procedure
+
+. Create a `ConfigMap` CR that describes the VLAN IDs for a group of cluster hosts. For example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: site-data
+  namespace: ztp-group
+  annotations:
+    argocd.argoproj.io/sync-options: Replace=true <1>
+data:
+  site-1-vlan: "101"
+  site-2-vlan: "234"
+----
+<1> The `argocd.argoproj.io/sync-options` annotation is required only if the `ConfigMap` is larger than 1 MiB in size.
++
+[NOTE]
+====
+The `ConfigMap` must be in the same namespace with the policy that has the hub template substitution.
+====
+
+. Commit the `ConfigMap` CR in Git, and then push to the Git repository being monitored by the Argo CD application.
+
+. Create a group PGT CR that uses a hub template to pull the required VLAN IDs from the `ConfigMap` object. For example, add the following YAML snippet to the group PGT CR:
++
+[source,yaml]
+----
+- fileName: SriovNetwork.yaml
+    policyName: "config-policy"
+    metadata:
+      name: "sriov-nw-du-mh"
+      annotations:
+        ran.openshift.io/ztp-deploy-wave: "10"
+    spec:
+      resourceName: du_mh
+      vlan: '{{hub fromConfigMap "" "site-data" (printf "%s-vlan" .ManagedClusterName) | toInt hub}}'
+----
+
+. Commit the group `PolicyGenTemplate` CR in Git, and then push to the Git repository being monitored by the Argo CD application.
++
+[NOTE]
+====
+Subsequent changes to the referenced `ConfigMap` CR are not automatically synced to the applied policies.
```

Comment:
The reader is left thinking this isn't possible. Can we link or make mention of the section below?